lib/instr.perf.ml

```diff
@@ -0,0 +1,16 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+let read_counter : unit -> (int, string) Result.t =
+  try
+    Perf.init ();
```

Comment:
In the current version, the instruction counter runs unconditionally from the initialization of this module, and the associated file descriptor is never collected (see `Perf.drop`). Not sure if that's a problem, the overhead might be very small.

Reply:
How silly of me: I forgot `Stdlib.at_exit` existed. Fixed.